### PR TITLE
fix: smtp send email fix for self hosted instance

### LINKED ISSUE #10592: Emails are not sent using smtp for self hosted instance
Found a bug? Please fill out the sections below. 👍

### Issue Summary

A summary of the issue. This needs to be a clear detailed-rich summary.

### Steps to Reproduce

1. Make a self hosted instance and use any smtp client by setting the env variables
2. Book any event and check if the booker is receiving an email or not

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Actual Results

Emails are not sent for a self hosted instance using smtp client

### Expected Results

Emails should be sent for a self hosted instance using smtp client

### PR DESCRIPTION
**Pull request by prateekjain98**
## What does this PR do?

Bug fix for sending email for a self hosted instance

Fixes #10592 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Create a self hosted instance and book any meeting and check if the emails are being sent correctly

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by Udit-takkar — COMMENTED** [review]
@prateekjain98 can you add some more information in the description about how this fixes the issue? 

sending emails worked fine for me locally without this change

**Comment by prateekjain98** [comment]
> @prateekjain98 can you add some more information in the description about how this fixes the issue?
> 
> sending emails worked fine for me locally without this change

It's working locally for me as well but on production, it does not. I was using gmail as my smtp client. You can use any smtp client on a deployed build and check.

File: packages/emails/templates/_base-email.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default class BaseEmail {
       ...(parseSubject.success && { subject: decodeHTML(parseSubject.data) }),
     };
 
-    new Promise((resolve, reject) =>
+    await new Promise((resolve, reject) =>
```
Comment (hariombalhara) [inline packages/emails/templates/_base-email.ts:54]: It might have caught this https://github.com/calcom/cal.com/issues/10398

**Review by hariombalhara — APPROVED** [review]
Thanks @prateekjain98 🙏 Appreciate the fix !!

**Review by zomars — APPROVED** [review]
Thank you for your contribution 🙏

**Comment by racso-dev** [comment]
Hi there, reopening this as I have the very same bug. I use resend as smtp, and it doesn't work. The smtp connection times out. 
```
2025-02-06T23:08:00.385171783Z @calcom/web:start: sendEmail from: calendar@tendiz.ai <calendar@tendiz.ai> subject: 30 Min Meeting between Oscar Rénier and Oscar Rénier Error: Connection timeout
2025-02-06T23:08:00.385181451Z @calcom/web:start:     at SMTPConnection._formatError (/calcom/node_modules/nodemailer/lib/smtp-connection/index.js:787:19)
2025-02-06T23:08:00.385190068Z @calcom/web:start:     at SMTPConnection._onError (/calcom/node_modules/nodemailer/lib/smtp-connection/index.js:773:20)
2025-02-06T23:08:00.385198624Z @calcom/web:start:     at Timeout.<anonymous> (/calcom/node_modules/nodemailer/lib/smtp-connection/index.js:232:22)
2025-02-06T23:08:00.385207621Z @calcom/web:start:     at listOnTimeout (node:internal/timers:569:17)
2025-02-06T23:08:00.385216036Z @calcom/web:start:     at process.processTimers (node:internal/timers:512:7) {
2025-02-06T23:08:00.385224432Z @calcom/web:start:   code: 'ETIMEDOUT',
2025-02-06T23:08:00.385232697Z @calcom/web:start:   command: 'CONN'
2025-02-06T23:08:00.385240972Z @calcom/web:start: }
```

My calcom is deployed through a docker container via coolify, and all the EMAIL_ env variables are properly set.

**Pull request merged**